src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/DataFrameTests.cs

```diff
@@ -50,6 +50,59 @@ public void TestCollect()
             Assert.Equal(19, row3.GetAs<int>("age"));
         }
 
+        [Fact]
+        public void TestWithColumn()
+        {
+            Func<Column, Column> sizeNameAgeUdf = Udf<Row, string>(
+                r =>
+                {
+                    string name = r.GetAs<string>("name");
+                    int? age = r.GetAs<int?>("age");
+                    if (age.HasValue)
+                    {
+                        return $"{r.Size()},{name},{age.Value}";
+                    }
+
+                    return $"{r.Size()},{name},{string.Empty}";
+                });
+
+            string[] allCols = _df.Columns().ToArray();
+            DataFrame nameAgeColDF =
+                _df.WithColumn("NameAgeCol", Struct(allCols[0], allCols.Skip(1).ToArray()));
+            DataFrame sizeNameAgeColDF =
+                nameAgeColDF.WithColumn("SizeNameAgeCol", sizeNameAgeUdf(nameAgeColDF["NameAgeCol"]));
+
+            Row[] originalDFRows = _df.Collect().ToArray();
+            Assert.Equal(3, originalDFRows.Length);
+
+            Row[] sizeNameAgeColDFRows = sizeNameAgeColDF.Collect().ToArray();
+            Assert.Equal(3, sizeNameAgeColDFRows.Length);
+
+            {
+                Row row = sizeNameAgeColDFRows[0];
+                Assert.Equal("Michael", row.GetAs<string>("name"));
+                Assert.Null(row.Get("age"));
+                Assert.Equal(originalDFRows[0], row.Get("NameAgeCol"));

```

Comment:
Can you also check the type of row.Get("NameAgeCol") so that it's easier to understand what type was expected/returned?

Reply:
Also, I guess `GetAs<Something>` is not possible?

Reply:
Added type check `Assert.IsType<Row>(row.Get("NameAgeCol"));` and also used `row.GetAs<Row>`